src/dune_rules/dep_conf_eval.ml

```diff
@@ -275,8 +275,11 @@ let named ~expander l =
 
 let unnamed ?(sandbox = Sandbox_config.no_special_requirements) ~expander l =
   let expander = prepare_expander expander in
-  ( List.fold_left l ~init:(Action_builder.return ()) ~f:(fun acc x ->
-        let+ () = acc
-        and+ _x = to_action_builder (dep expander x) in
-        ())
+  ( (let+ paths =
+       List.fold_left l ~init:(Action_builder.return []) ~f:(fun acc x ->
+           let+ acc = acc
+           and+ paths = to_action_builder (dep expander x) in
+           paths :: acc)
+     in
+     List.concat paths)
```

Comment:
Doesn't seem good to collect all these paths everywhere if they're only used for melange. What about adding a separate function?

Reply:
added in [e679967](https://github.com/ocaml/dune/pull/7199/commits/e679967535f2f426aa268a971c6b99674ce4cb6a)